SrcMLCppAPI/SrcMLCppAPI.cpp

```diff
@@ -3,10 +3,45 @@
 #include "stdafx.h"
 #include "srcml.h"
 #include "SrcMLCppAPI.h"
-
+#include <iostream>
+#include <string>
 using namespace System;
 using namespace System::Runtime::InteropServices;
+void SetMetaData(srcml_unit* unit, SourceData* archiveData){
```

Comment:
@patrickfrancis I'm still heavily debating what I should do with SetMetaData. Originally, it was meant to make it easy to mass set all of the options but some options are on the archive, some are on unit, etc. I don't exactly like it but I have to think about what to do with it.

As for thread safety, if you're threading at the method/function level and writing to separate archives, then it should be fine (though I'll write a test or something to make sure this is true). If you're trying to write to the same archive, then maybe not. But in that last case, I'm pretty sure srcML would have to be the one doing the threading, which I believe it does. I'd have to look and see if there's a way to instruct it to do so since I'm not sure it'll do it automatically from memory buffers.